src/main/java/homework_5/custom_regex_matcher/CustomRegexMatcher.java

```diff
@@ -0,0 +1,23 @@
+package homework_5.custom_regex_matcher;
+
+import homework_5.AppHW_5;
+
+public class CustomRegexMatcher extends AppHW_5 {
+    private final static String PATTERN_STRING = "[0-9a-zA-Z_]+@[0-9a-zA-Z_]+\\.[a-zA-Z]{2,3}";
+
+    public CustomRegexMatcher() {
+        super(PATTERN_STRING);
+    }
+
+    public String run() {
+        readData();
+        String result = String.valueOf(check());
+        System.out.println(result);
+        closeReader();
+        return result;
```

Comment:
Opt: as example here it's pretty unclear what is happening. Void methods usually are not recommended too